Source/Core/Common/FileUtil.cpp

```diff
@@ -276,33 +279,68 @@ bool DeleteDir(const std::string& filename)
   return false;
 }
 
+// Repeatedly invokes func until it returns true or max_attempts failures.
+// Waits after each failure, with each delay doubling in length.
+template <typename FuncType>
+static bool AttemptMaxTimesWithExponentialDelay(const FuncType& func, int max_attempts,
+                                                std::chrono::milliseconds delay)
+{
+  for (int failed_attempts = 0; failed_attempts < max_attempts; ++failed_attempts)
+  {
+    if (func())
+    {
+      return true;
+    }
+    if (failed_attempts + 1 < max_attempts)
+    {
+      INFO_LOG_FMT(COMMON, "Rename attempt failed, delaying for {} milliseconds", delay.count());
+      std::this_thread::sleep_for(delay);
+      delay *= 2;
+    }
+  }
+  return false;
+}
+
 // renames file srcFilename to destFilename, returns true on success
 bool Rename(const std::string& srcFilename, const std::string& destFilename)
 {
   INFO_LOG_FMT(COMMON, "Rename: {} --> {}", srcFilename, destFilename);
 #ifdef _WIN32
-  auto sf = UTF8ToTStr(srcFilename);
-  auto df = UTF8ToTStr(destFilename);
-  // The Internet seems torn about whether ReplaceFile is atomic or not.
-  // Hopefully it's atomic enough...
-  if (ReplaceFile(df.c_str(), sf.c_str(), nullptr, REPLACEFILE_IGNORE_MERGE_ERRORS, nullptr,
-                  nullptr))
-    return true;
-  // Might have failed because the destination doesn't exist.
-  if (GetLastError() == ERROR_FILE_NOT_FOUND)
-  {
-    if (MoveFile(sf.c_str(), df.c_str()))
+  // Store converted strings in local variables so cstrings don't point to temporary objects
```

Comment:
This shouldn't be required. source_wstring and destination_wstring won't go out of scope and won't get destructed until `attempt_func` completes, so there is no lifetime issue here.

That said, the original naming is pretty terrible so I guess we can keep the new names but get rid of source_ptr and destination_ptr.

Reply:
The comment was referring to why I created source_wstring instead of assigning source_ptr = UTF8ToTStr(srcFilename).c_str(), but given the spacing I see how that's ambiguous.  Getting rid of source_ptr makes the comment redundant anyway so I'll get rid of it.